File: sentimental-credit/credit.py
from cs50 import get_string

number = get_string("Number: ")
num = int(number)
length = len(number)

def legit(number):
    total_sum = 0
    digits = list(number)
    length = len(number)

    # Starting from the rightmost digit (working right to left)
    for i in range(length-1, -1, -1):  # Count backwards from last digit
        digit = int(digits[i])

        # Every other digit starting from second-to-last
        if (length - i) % 2 == 0:
            # Double the digit
            doubled = digit * 2
            # If doubled digit is > 9, sum its digits (or subtract 9, same result)
            if doubled > 9:
                doubled -= 9
            total_sum += doubled
        else:
            total_sum += digit

    if total_sum % 10 != 0:
        print("INVALID")
        return 1
    else:
        return 0


if legit(number) == 0 :
    if len(number) == 15:
        if int(number[:2]) == 34 or int(number[:2]) == 37:
            print ("AMEX")
        else:
            print("INVALID")
    elif len(number) == 16:
        if int(number[:2]) in [51,55]:
            print ("MASTERCARD")
        if int(number[0]) == 4:
            print ("VISA")
        else:
            print("INVALID")
    elif len(number) == 13 and int(number[0]) == 4:
        print ("VISA")
    else:
        print("INVALID")

File: credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long number;
    do {
        number = get_long("Credit card number: ");
    }
    while (number<0);

    long original = number;
    int count = 1;
    int sum1 = 0;
    int sum2 = 0;
    int digit;

    while (number > 0) {
        digit = number % 10;  // Get the last digit
        // printf("%d(%i) ", digit, count);     // Print the digit

        if (count%2 == 0){
            int new_d1;
            if (digit>=5){
                new_d1 = ((digit * 2) % 10) + 1;}
            else {new_d1 = digit*2;}

        sum1 = sum1 + new_d1;}

        else {sum2 += digit;}

        count ++;
        number = number / 10;}

    int total = sum1+sum2;
    count = count -1;


    if (total % 10 == 0) {
        long first_two_digits = original;
        while (first_two_digits >= 100) {
            first_two_digits /= 10;
        }
        // printf("first2: %li\n", first_two_digits);
        if ((first_two_digits==34||first_two_digits==37) && count ==15){
            printf("AMEX\n");}
        else if (digit == 4 && (count == 13||count == 16)){
            printf("VISA\n");}
        else if (first_two_digits >= 51 && first_two_digits <= 55 && count ==16){
            printf("MASTERCARD\n");}
        else {
            printf("INVALID\n");}
    }
    else {printf("INVALID\n");}
}